server/src/main/java/org/elasticsearch/search/vectors/KnnSearchBuilder.java

```diff
@@ -123,6 +167,7 @@ public boolean equals(Object o) {
             && numCands == that.numCands
             && Objects.equals(field, that.field)
             && Arrays.equals(queryVector, that.queryVector)
+            && Objects.equals(filterQueries, that.filterQueries)
```

Comment:
Should we add `filterQueries` to `hashCode()` as well?

Reply:
Yes, thanks for catching that!